### PR TITLE
docs: document node groups and migration from old config fields

### PR DESCRIPTION
**Pull request by malt3**
### Context

As part of the migration to node groups ([AB#3100](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3100)), we need to document the changes for the end user that were introduced by #2152.

### Proposed change(s)

- docs: document node groups and migration from old config fields

### Checklist

- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: docs/docs/reference/migration.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ Use [`constellation config migrate`](./cli.md#constellation-config-migrate) to a
 ## Migrating from CLI versions before 2.10
 
 - AWS cluster upgrades require additional IAM permissions for the newly introduced `aws-load-balancer-controller`. Please upgrade your IAM roles using `iam upgrade apply`. This will show necessary changes and apply them, if desired.
+- The global `nodeGroups` field was added.
+- The fields `instanceType`, `stateDiskSizeGB` and `stateDiskType` for each cloud provider are now part of the configuration of individual node groups.
```
Comment (thomasten) [inline docs/docs/reference/migration.md:21]:
```suggestion
- The fields `instanceType`, `stateDiskSizeGB`, and `stateDiskType` for each cloud provider are now part of the configuration of individual node groups.
```
MS style guide recommends Oxford comma

File: docs/docs/workflows/config.md [hunk 2]
```diff
@@ -80,6 +80,50 @@ SNP-based attestation will be enabled as soon as a fix is verified.
 
 Fill the desired VM type into the **instanceType** fields in the `constellation-conf.yml` file.
 
+## Creating additional node groups
+
+By default, Constellation will create the node groups `control_plane_default` and `worker_default` for control-plane nodes and workers respectively.
```
Comment (thomasten) [inline docs/docs/workflows/config.md:85]:
```suggestion
By default, Constellation creates the node groups `control_plane_default` and `worker_default` for control-plane nodes and workers, respectively.
```
Not from the style guide we follow, but IMO still good recommendation: "Use present tense for statements that describe general behavior that's not associated with a particular time." (https://developers.google.com/style/tense)

File: docs/docs/workflows/config.md [hunk 3]
```diff
@@ -80,6 +80,50 @@ SNP-based attestation will be enabled as soon as a fix is verified.
...
+If you require additional control-plane or worker groups with different instance types, zone placements or disk sizes, you can add additional node groups to the `constellation-conf.yml` file.
```
Comment (thomasten) [inline docs/docs/workflows/config.md:86]:
```suggestion
If you require additional control-plane or worker groups with different instance types, zone placements, or disk sizes, you can add additional node groups to the `constellation-conf.yml` file.
```

File: docs/docs/workflows/config.md [hunk 4]
```diff
@@ -80,6 +80,50 @@ SNP-based attestation will be enabled as soon as a fix is verified.
...
+Each node group can be scaled individually.
```
Comment (thomasten) [inline docs/docs/workflows/config.md:87]: You already updated the "Scale your cluster" workflow some time ago. Is it still correct for the final implementation?

File: docs/docs/workflows/config.md [hunk 5]
```diff
@@ -80,6 +80,50 @@ SNP-based attestation will be enabled as soon as a fix is verified.
...
+    instanceType: c6a.24xlarge
+    stateDiskSizeGB: 128
+    stateDiskType: gp3
+    zone: eu-west-1c
+    initialCount: 1
+```
+
+This configuration creates an additional node group `high_cpu` with a larger instance type and disk.
+
+The field `zone` can be used to specify what availability zone nodes of the group are placed in.
```
Comment (thomasten) [inline docs/docs/workflows/config.md:119]:
```suggestion
You can use the field `zone` to specify what availability zone nodes of the group are placed in.
```
prefer active voice

**Review by thomasten — COMMENTED** [review]
Looks good. Some nits and one question.

File: docs/docs/workflows/config.md (hunk 4 above)
Comment (malt3) [reply docs/docs/workflows/config.md:87]: Yes. The autoscaling docs already mention node groups and explain how users can choose a node group / scaling group to configure the autoscaling settings.

**Review by thomasten — APPROVED** [review]

**Pull request merged**